docs/installation.rst

```diff
@@ -76,3 +76,48 @@ Please add the following to your ``settings.py``:
 
 Finally let your OpenID connect OP know about your callback URL. In our example this is:
 ``http://127.0.0.1:8000/oidc/callback/``.
+
+
+Additional optional configuration
+=================================
+
+Connecting OIDC user identities to Django users
+-----------------------------------------------
+
+By default, mozilla-django-oidc looks up a Django user matching the email field
+to the email address returned in the user info data from the OIDC provider.
+
+This means that no two users in the Django user table can have the same email
+address. Since the email field is not unique, it's possible that this can
+happen. Especially if you allow users to change their email address. If it ever
+happens, then the users in question won't be able to authenticate.
+
+If you want different behavior, subclass the
+:py:class:`mozilla_django_oidc.auth.OIDCAuthenticationBackend` class and
+override the `filter_users_by_claims` method.
+
+For example, let's say we store the email address in a ``Profile`` table
+in a field that's marked unique so multiple users can't have the same
+email address. Then we could do this:
+
+.. code-block:: python
+
+   from mozilla_django_oidc.auth import OIDCAuthenticationBackend
+
+   class MyOIDCAB(OIDCAuthenticationBackend):
+       def filter_users_by_claims(self, claim):
```

Comment:
Part of me was wondering whether we should talk about what's in the claim and/or tell the user/developer how to figure out what's in the claim. I'm pretty sure the information in the claim is different depending on whether you're using Mozilla's SSO or Auth0 or some other OIDC provider, so I'm not sure how to do this effectively. Maybe have a paragraph after this along the lines of:

> The information available in the claim depends on which OIDC provider you're using. Best way to figure out what's in the claim is to log the information in a development environment.

Maybe we should also add?:

> The data in the claim isn't static and can change over time and also be different depending on the user that logged in.

Are both of those true in your experience? Should we add them to reduce the surprises users/developers might encounter when trying to integrate mozilla-django-oidc into their sites?